circleci/catapult

```diff
@@ -4,21 +4,27 @@
 #
 # Usage:
 #
-#   circleci-catapult [CATAPULT_URL] [APP_NAME]
+#   circleci-catapult [APP_NAME]
 
-# TODO: Implement functionality from here...
-# https://github.com/Clever/drone/blob/master/plugin/publish/catapult.go
 set -e
 
 # User supplied args
-CATAPULT_URL=$1
-if [[ -z $CATAPULT_URL ]]; then echo "Missing arg1 CATAPULT_URL" && exit 1; fi
-APP_NAME=$2
+APP_NAME=$1
 if [[ -z $APP_NAME ]]; then echo "Missing arg2 APP_NAME" && exit 1; fi
 
 # Set automatically by CircleCI
 : ${CIRCLE_PROJECT_REPONAME?"Missing required env var"}
 REPO=$CIRCLE_PROJECT_REPONAME
+: ${CIRCLE_PROJECT_USERNAME?"Missing required env var"}
+USER=$CIRCLE_PROJECT_USERNAME
+: ${CIRCLE_BUILD_NUM?"Missing required env var"}
+BUILD_NUM=$CIRCLE_BUILD_NUM
 
-echo "Error! Not yet implemented"
-return 1
+# Should be in CircleCI
+: ${CATAPULT_URL?"Missing required env var"}
+CATAPULT_URL=$CATAPULT_URL
+: ${CATAPULT_BASIC_AUTH?"Missing required env var"}
+BASIC_AUTH=$CATAPULT_BASIC_AUTH
```

Comment:
For other commands, we've been passing all data via args. These args, if they're secret, use circle's env vars. 

Whether we use args or env vars, I think it will require updating the `circle.yml` if we change our interface.

--

Some options:

(1) As you've configured it:

```
CATAPULT_URL=$CATAPULT_URL CATAPULT_BASIC_AUTH=$CATAPULT_BASIC_AUTH ./circleci/catapult app-name
```

(2) Use only args (requires being careful about arg order):

```
./circleci/catapult $CATAPULT_APP $CATAPULT_URL $CATAPULT_AUTH
```

(3) pass env vars, but make env var names simple here in script:

```
APP=$CATAPULT_APP URL=$CATAPULT_URL AUTH=$CATAPULT_AUTH ./circleci/catapult
```

(4) pass everything via env vars, use full names:

```
CATAPULT_APP=some-app CATAPULT_URL=some-url CATAPULT_AUTH=$CATAPULT_AUTH ./circleci/catapult
```